diff --git a/compiler/systems/t_linux.pas b/compiler/systems/t_linux.pas
index 2b53351..a42ddbf 100644
--- a/compiler/systems/t_linux.pas
+++ b/compiler/systems/t_linux.pas
@@ -186,9 +186,9 @@ begin
 
 {$ifdef arm}
 {$ifdef FPC_ARMEL}
-     defdynlinker:='/lib/ld-linux.so.3';
+     defdynlinker:='/lib/ld-linux-armhf.so.3';
 {$else FPC_ARMEL}
-     defdynlinker:='/lib/ld-linux.so.2';
+     defdynlinker:='/lib/ld-linux-armhf.so.3';
 {$endif FPC_ARMEL}
 {$endif arm}
 
